drivers/usb/udc/udc_it82xx2.c

```diff
@@ -0,0 +1,1481 @@
+				     config->ep_cfg_in[0].caps.mps, 0);
+	if (ret) {
+		LOG_ERR("Failed to enable ep 0x%02x", USB_CONTROL_EP_IN);
+		return ret;
+	}
+	return 0;
+}
+
+static int it82xx2_shutdown(const struct device *dev)
+{
+	if (udc_ep_disable_internal(dev, USB_CONTROL_EP_OUT)) {
+		LOG_ERR("Failed to disable control endpoint");
+		return -EIO;
+	}
+
+	if (udc_ep_disable_internal(dev, USB_CONTROL_EP_IN)) {
+		LOG_ERR("Failed to disable control endpoint");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int it82xx2_lock(const struct device *dev)
+{
+	return udc_lock_internal(dev, K_FOREVER);
+}
+
+static int it82xx2_unlock(const struct device *dev)
+{
+	return udc_unlock_internal(dev);
+}
+
+static const struct udc_api it82xx2_api = {
+	.ep_enqueue = it82xx2_ep_enqueue,
+	.ep_dequeue = it82xx2_ep_dequeue,
+	.ep_set_halt = it82xx2_ep_set_halt,
+	.ep_clear_halt = it82xx2_ep_clear_halt,
+	.ep_try_config = NULL,
+	.ep_enable = it82xx2_ep_enable,
+	.ep_disable = it82xx2_ep_disable,
+	.host_wakeup = it82xx2_host_wakeup,
+	.set_address = it82xx2_set_address,
+	.enable = it82xx2_enable,
+	.disable = it82xx2_disable,
+	.init = it82xx2_init,
+	.shutdown = it82xx2_shutdown,
+	.lock = it82xx2_lock,
+	.unlock = it82xx2_unlock,
+};
+
+static int it82xx2_usb_driver_preinit(const struct device *dev)
+{
+	const struct usb_it82xx2_config *config = dev->config;
+	struct udc_data *data = dev->data;
+	struct it82xx2_data *priv = udc_get_private(dev);
+	int err;
+
+	k_mutex_init(&data->mutex);
+	k_fifo_init(&priv->fifo);
+
+	err = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
+	if (err < 0) {
+		LOG_ERR("Failed to configure usb pins");
+		return err;
+	}
+
+	for (int i = 0; i < MAX_NUM_ENDPOINTS; i++) {
+		config->ep_cfg_out[i].caps.out = 1;
+		if (i == 0) {
+			config->ep_cfg_out[i].caps.control = 1;
+			config->ep_cfg_out[i].caps.mps = USB_CONTROL_EP_MPS;
+		} else if ((i % 3) == 2) {
+			config->ep_cfg_out[i].caps.bulk = 1;
+			config->ep_cfg_out[i].caps.interrupt = 1;
+			config->ep_cfg_out[i].caps.iso = 1;
+			config->ep_cfg_out[i].caps.mps = 64;
+		}
+
+		config->ep_cfg_out[i].addr = USB_EP_DIR_OUT | i;
+		err = udc_register_ep(dev, &config->ep_cfg_out[i]);
+		if (err != 0) {
+			LOG_ERR("Failed to register endpoint");
+			return err;
+		}
+	}
+
+	for (int i = 0; i < MAX_NUM_ENDPOINTS; i++) {
+		config->ep_cfg_in[i].caps.in = 1;
+		if (i == 0) {
+			config->ep_cfg_in[i].caps.control = 1;
+			config->ep_cfg_in[i].caps.mps = USB_CONTROL_EP_MPS;
+		} else if ((i % 3) != 2) {
+			config->ep_cfg_in[i].caps.bulk = 1;
+			config->ep_cfg_in[i].caps.interrupt = 1;
+			config->ep_cfg_in[i].caps.iso = 1;
+			config->ep_cfg_in[i].caps.mps = 64;
+		}
+
+		config->ep_cfg_in[i].addr = USB_EP_DIR_IN | i;
+		err = udc_register_ep(dev, &config->ep_cfg_in[i]);
+		if (err != 0) {
+			LOG_ERR("Failed to register endpoint");
+			return err;
+		}
+	}
+
+	data->caps.rwup = true;
+	data->caps.hs = false;
+	data->caps.out_ack = false;
```

Comment:
you can remove it, no need to initialize static variables with false.

Reply:
Done